File: dialogplus/src/test/java/com/orhanobut/android/dialogplus/ViewHolderTest.java
package com.orhanobut.android.dialogplus;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.LinearLayout;

import com.orhanobut.dialogplus.BuildConfig;
import com.orhanobut.dialogplus.Holder;
import com.orhanobut.dialogplus.ViewHolder;

import junit.framework.TestCase;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.Robolectric;
import org.robolectric.RobolectricGradleTestRunner;
import org.robolectric.annotation.Config;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * @author Orhan Obut
 */
@RunWith(RobolectricGradleTestRunner.class)
@Config(constants = BuildConfig.class, sdk = 18)
public class ViewHolderTest extends TestCase {

  private final Context context;

  public ViewHolderTest() {
    context = Robolectric.buildActivity(Activity.class).create().get();
  }

  private ViewHolder getHolder() {
    ViewHolder holder = new ViewHolder(new LinearLayout(context));
    LayoutInflater layoutInflater = LayoutInflater.from(context);
    holder.getView(layoutInflater, new LinearLayout(context));
    return holder;
  }

  @Test
  public void init() {
    assertThat(getHolder()).isInstanceOf(Holder.class);
    assertThat(getHolder()).isNotNull();
  }

  @Test
  public void testViewInflation() {
    View contentView = new LinearLayout(context);
    ViewHolder holder = new ViewHolder(contentView);
    LayoutInflater layoutInflater = LayoutInflater.from(context);
    View view = holder.getView(layoutInflater, new LinearLayout(context));

    assertThat(view).isNotNull();
    assertThat(holder.getInflatedView()).isEqualTo(contentView);
  }

  @Test
  public void testFooter() {
    ViewHolder holder = getHolder();

    assertThat(holder.getFooter()).isNull();

    View footer = new LinearLayout(context);
    holder.addFooter(footer);

    assertThat(holder.getFooter()).isEqualTo(footer);
  }

  @Test
  public void testHeader() {
    ViewHolder holder = getHolder();

    assertThat(holder.getHeader()).isNull();

    View header = new LinearLayout(context);
    holder.addHeader(header);

    assertThat(holder.getHeader()).isEqualTo(header);
  }

}

File: dialogplus/src/test/java/com/orhanobut/android/dialogplus/ViewHolderTest.kt
package com.orhanobut.android.dialogplus

import android.app.Activity
import android.view.LayoutInflater
import android.widget.LinearLayout
import com.orhanobut.dialogplus.Holder
import com.orhanobut.dialogplus.ViewHolder
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.Robolectric
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class ViewHolderTest {

  private val context = Robolectric.setupActivity(Activity::class.java)

  private val holder: ViewHolder
    get() {
      val holder = ViewHolder(LinearLayout(context))
      val layoutInflater = LayoutInflater.from(context)
      holder.getView(layoutInflater, LinearLayout(context))
      return holder
    }

  @Test fun init() {
    assertThat(holder).isInstanceOf(Holder::class.java)
    assertThat(holder).isNotNull()
  }

  @Test fun testViewInflation() {
    val contentView = LinearLayout(context)
    val holder = ViewHolder(contentView)
    val layoutInflater = LayoutInflater.from(context)
    val view = holder.getView(layoutInflater, LinearLayout(context))

    assertThat(view).isNotNull()
    assertThat(holder.inflatedView).isEqualTo(contentView)
  }

  @Test fun testFooter() {
    val holder = holder

    assertThat(holder.footer).isNull()

    val footer = LinearLayout(context)
    holder.addFooter(footer)

    assertThat(holder.footer).isEqualTo(footer)
  }

  @Test fun testHeader() {
    val holder = holder

    assertThat(holder.header).isNull()

    val header = LinearLayout(context)
    holder.addHeader(header)

    assertThat(holder.header).isEqualTo(header)
  }

}
